pkg/statsd/receiver.go

```diff
@@ -121,3 +123,27 @@ func getIP(addr net.Addr) gostatsd.IP {
 	log.Errorf("Cannot get source address %q of type %T", addr, addr)
 	return gostatsd.UnknownIP
 }
+
+// bufferPool is a strongly typed wrapper around a sync.Pool for [][]byte
+type bufferPool struct {
+	p sync.Pool
+}
+
+func newBufferPool() *bufferPool {
+	return &bufferPool{
+		p: sync.Pool{
+			New: func() interface{} {
+				b := [][]byte{make([]byte, packetSizeUDP)}
+				return &b
```

Comment:
Why a pointer?

Reply:
gometalinter says to use a pointer to avoid a copy.  I'm not 100% certain the need myself, but it doesn't seem to make much difference either way.  I think if I made the wrapping pool a *[][]byte and pushed some extra work in to the doneFunc I could remove it, but the whole point of the wrapper is to not deal with that.

Reply:
I think it is a bug in metalinter. Pointer is 1 machine word, slice is 3. It is more expensive to put that 3-word slice on the heap than to copy it.

Reply:
True story.  Looks like it just makes the copy explicit instead of implicit.  Will update.